### PR TITLE
#156 Hashing implementation

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Closes: #156

@yegor256 WDYT about such algorithm?

The results: 
- `hash TRUE` = 72057594037927944
- `hash FALSE` = 8
- `hash 42` = 3026418949609750528
- `hash 14325.2` = 1300271141437013430
- `hash "Hello world"` = 7535155495499350018

### CODE REVIEW

**Comment by yegor256** [comment]
@maxonfjvipon I love it, but would be great if you can add a comment/doc to the `hash` object. Now it's not clear what is it, how it works, and what it's for.

**Comment by maxonfjvipon** [comment]
@yegor256 sure, btw, maybe the proper name would be `hash-code` (or `hash-code-of`) because hash is usually string

**Comment by yegor256** [comment]
Screenshot 2023-03-01 at 17 16 07

@maxonfjvipon let's call it `hash-code-of`

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look, something's wrong with build

**Comment by maxonfjvipon** [comment]
@yegor256 previous hashing implementation for some reason takes about 30 seconds on my laptop to calculate hash code of just number. That's too much so I kind of copied and united current hashing implementations from `eo-runtime`

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo is there the same problem as was in `eo-runtime`

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo seems your fix did not help

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon because @yegor256 returned the bug back - https://github.com/objectionary/eo-collections/pull/161 :) 

Ok, will fix rultor soon and hope it won't happen again. You can check progress [here](https://github.com/yegor256/rultor/issues/1237).

**Comment by yegor256** [comment]
@volodya-lombrozo it's not me, it's [my script](https://github.com/yegor256/mergem) :)  Find a way to prevent this bot to submit pull requests with this particular change. There has to be a way.

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon Could you try to update that PR, please?

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon I'm afraid it will fail. Could you repeat that changes, please? https://github.com/objectionary/eo-collections/pull/160/files

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo it seems windows build still does not work

**Comment by maxonfjvipon** [comment]
@yegor256 please, have a look

File: src/test/eo/org/eolang/collections/hash-code-of-tests.eo [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      hash-code-of str
+
+[] > hash-code-of-different-numbers-are-different
+  assert-that > @
+    hash-code-of 42
+    $.not
+      $.equal-to
+        hash-code-of 24
+
+[] > hash-of-int-is-int
```
Comment (volodya-lombrozo) [inline src/test/eo/org/eolang/collections/hash-code-of-tests.eo:63]: @maxonfjvipon Please, move that test near to the `hash-code-bools-is-integer`

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]
@maxonfjvipon Can we add more tests for that object? It would be interesting to test floating numbers and, maybe, big numbers (or some corner cases).

File: src/test/eo/org/eolang/collections/hash-code-of-tests.eo (hunk 1 above)
Comment (maxonfjvipon) [reply src/test/eo/org/eolang/collections/hash-code-of-tests.eo]: Done

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo I moved the test and added `todo` to add more tests in the future. WDYT?

**Review by volodya-lombrozo — APPROVED** [review]
@maxonfjvipon Looks good to me. Thanks.

**Comment by maxonfjvipon** [comment]
@yegor256 please merge

**Comment by yegor256** [comment]
@rultor merge
